### DISCUSSION TITLE
We may drop v3 support in Lavalink v4

### DISCUSSION THREAD

**Discussion by freyacodes** [lavalink-devs/Lavalink#859] [Announcements]
[Written by @TopiSenpai]

hey peps,
Originally we were planning to make Lavalink v4 support the v3(old) & v4(new) API.
At this point we are unsure if this is really a good idea & what people need/want.
We want to ask you about feedback what you think would be the best approach. Here are some pros & cons to support the v3 API in v4

Lets go over the pros of supporting both versions at once:
  - there is one codebase to maintain and v3 clients can just update to the new v4 version without needing to do any changes
  - we can guarantee Lavaplayer updates to the v3 API without issue
  - we can freeze v3 development entirely

There are some cons too:
  - it may be confusing for lib devs/end users which version of Lavalink to use
  - we would only keep support for the v3.7 API which means you would need to update the rest endpoints regardless if you use the old v3 protocol
  - handling both v3 & v4 protocol in one version increases complexity in handling propper behaviour and can lead to more bugs/less maintainable code or slower updates

Arguably it would be just easier to develop v4 independantly from v3 and provide Lavaplayer as long as we can. 
The reason why we initially decided on this was because there are A LOT of bots & clients which are not being updated and still want to use the newest Lavalink version. We can't guarantee constant Lavaplayer updates for v3.

If you have any feedback/ideas about this let us know in the *Lavalink v4 discussion* on our [Discord server](https://discord.gg/ZW4s47Ppw4)

**Comment by topi314** [comment]
we decided to go ahead and not provide both `/v3` & `/v4` in lavalink v4
instead we will keep `v3` updated & backport bug-fixes
